Implemented Merge Sort Algorithm

sharanyaa23 opened the pull request:
## Fixes #182 

## ✅ **Contribution Checklist (make sure all are checked)**

- [x] Code file added (`.cpp`, `.java`, `.py`, etc.)
- [ ] Markdown file added with:
  - [x] Problem statement  
  - [x] Example input/output  
  - [x] Generic logic  
  - [x] Time & space complexity  

---

## 📝 **Type of Addition (check any one)**

- [x] DSA problem addition  
- [ ] Fix mistakes in existing codebase
- [ ] Add better solutions  
- [ ] Make docs better  

---

## 📊 **Last check** (make sure all are checked

- [x] **Have you checked that this problem isn't already in the codebase?🔍**
- [x] Did you cross-check that your code/explanation is right?

## 🏢 **Your Organization?**
- [ ] GsSOC'25

shivamm-verma commented:
Do not perform DELETIONS @sharanyaa23 
I will perform the deletion, kindly `git pull origin main` and just add your new contribution of merge sort into it.
### I suggest raising a new PR for that

sharanyaa23 commented:
@shivamm-verma there was a duplicate folder of Sorting-Algos that is why I deleted it. Also the existing java and c++ merge-sort algorithm was giving some issues. I corrected them and also added necessary comments wherever required. Should I raise a PR again or no?
